compat/memset_s.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_COMPAT_MEMSET_S_H
+#define SQUID_COMPAT_MEMSET_S_H
+
+#if !HAVE_MEMSET_S
+
+typedef int errno_t;
+typedef size_t rsize_t;
+
+/**
+ * to zero a buffer in a more secure manner meant for a handful of purposes.
+ * e.g. for password matters.
+ * The compiler can optimize away a memset call to gain performance here
+ * making sure it does not occur.
```

Comment:
Re-documenting a standard function is a bad idea IMO -- our custom documentation will either pointlessly duplicate the official one or miss important caveats. For example, compilers can only remove _some_ memset() calls while the above custom documentation can be misinterpreted as implying that they can remove _any_ memset() call. This compat file is meant to replace a standard memset_s(). We should not document what a standard memset_s() is.

Reply:
Yes you re right.